### PR TITLE
Fix flaky e2e test

### LINKED ISSUE #56636: native query params not cleaned from URL after leaving editor - race condition on slow network
Is it an edge case of https://github.com/metabase/metabase/issues/52806

if you simulate "fast 4g" network, the repro test is failing almost always locally, location is changed to correct and immediately after it is changed to the wrong one.

Image

causes flakiness on `should remove parameter values from the URL when leaving the query builder and discarding changes (metabase#52806)` spec

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/56636
Stress test (100x, slow network) https://github.com/metabase/metabase/actions/runs/17467374403/job/49606504035

### CODE REVIEW

File: e2e/test/scenarios/native/native-reproductions.cy.spec.ts [hunk 1]
```diff
@@ -396,19 +396,16 @@ describe("issue 52806", () => {
...
-      H.modal().button("Discard changes").click();
-      cy.findByTestId("home-page");
-      cy.location().should((location) => expect(location.search).to.eq(""));
-    },
-  );
+  it("should remove parameter values from the URL when leaving the query builder and discarding changes (metabase#52806)", () => {
+    cy.visit("/");
+    H.newButton("SQL query").click();
+    H.NativeEditor.focus().type("select {{x}}");
+    cy.location().should((location) => expect(location.search).to.eq("?x="));
```
Comment (ranquild) [inline e2e/test/scenarios/native/native-reproductions.cy.spec.ts:403]: The only new line

**Review by romeovs — APPROVED** [review]
